### PR TITLE
[MM-62024] Update server dependencies to use github.com/mattermost/mattermost/server/v8

### PR DESCRIPTION
**Pull request by hanzei**
## Summary
In preparation for https://github.com/mattermost/mattermost-plugin-playbooks/pull/1957, this PR updates the go dependencies to the modern era of `github.com/mattermost/mattermost/server/public`.

The changes are quite straightforward. It's mostly import path changes, a bit of cleanup of deprecated methods, and fixing method calls where the arguments changed.

## Ticket Link
https://mattermost.atlassian.net/browse/MM-62024

## Checklist

- [x] ~~Telemetry updated~~
- [x] ~~Gated by experimental feature flag~~
- [x] ~~Unit tests updated~~

### CODE REVIEW

File: server/app/playbook_service.go [hunk 1]
```diff
@@ -24,6 +25,12 @@ type playbookService struct {
 	metricsService *metrics.Metrics
 }
 
+type InsightsOpts struct {
+	StartUnixMilli int64
+	Page           int
+	PerPage        int
+}
+
```
Comment (lieut-data) [inline server/app/playbook_service.go:33]: Not necessary for this PR, but I wonder if we can remove all the associated code connected with insights?

**Review by lieut-data — APPROVED** [review]
Thanks, @hanzei! This looks solid :)

**Review by agarciamontoro — COMMENTED** [review]
All looks good! I only have one question: Why do we need the new `build/pluginctl/logs.go` file? Is that something required for the update?

**Comment by hanzei** [comment]
@agarciamontoro I synced this repo with https://github.com/mattermost/mattermost-plugin-starter-template. I recently added the `make logs` command to  that repo.

**Review by agarciamontoro — APPROVED** [review]
Makes sense, @hanzei. Thank you!

**Comment by hanzei** [comment]
@lindy65 Would you be the correct QA person to test playbooks PRs?

**Comment by hanzei** [comment]
Thanks, @lindy65, I opened https://mattermost.atlassian.net/browse/MM-62024.

**Pull request merged**